conans/client/remover.py

```diff
@@ -168,7 +168,7 @@ def remove(self, pattern, remote_name, src=None, build_ids=None, package_ids_fil
             else:
                 refs = self._remote_manager.search_recipes(remote, pattern)
         else:
-            if input_ref:
+            if input_ref and check_valid_ref(input_ref, allow_pattern=False):
```

Comment:
Could this be simplified to...
```suggestion
            if input_ref and check_valid_ref(input_ref, allow_pattern=False):
```
this?

Reply:
nope, because only when input_ref is valid, check_valid_ref validates "*" in the reference. Otherwise, check_valid_ref will try to create a new reference obj with input_ref. I tried this approach on last commit, but didn't work.

Reply:
I understand the issue but to me having ``check_valid_ref(allow_pattern=False)`` to check for a valid reference that contains `*` makes me think the opposite as I read the code. Maybe this works?

```suggestion
            if check_valid_ref(input_ref, allow_pattern=True):
```

Reply:
no, because `check_valid_ref` checks first `input_ref`, and when is None it won't check `*`. The `allow_pattern` only affects when `input_ref` is a valid reference.